Index: trunk/src/VBox/ExtPacks/VBoxDTrace/VBoxDTraceR0.cpp
===================================================================
--- trunk/src/VBox/ExtPacks/VBoxDTrace/VBoxDTraceR0.cpp	(revision 58986)
+++ trunk/src/VBox/ExtPacks/VBoxDTrace/VBoxDTraceR0.cpp	(revision 59959)
@@ -1271,5 +1271,5 @@
     AssertReturnVoid(pMtx != &g_DummyMtx);
     AssertPtr(pMtx);
-    if (pMtx->hMtx == NIL_RTSEMMUTEX || pMtx->hMtx == NULL)
+    if (pMtx->hMtx == NIL_RTSEMMUTEX)
         return;
 
