--- VirtualBox-5.1.4/src/VBox/Additions/linux/drm/vbox_drv.c.orig	2016-09-10 04:31:04.190301372 +0100
+++ VirtualBox-5.1.4/src/VBox/Additions/linux/drm/vbox_drv.c	2016-09-10 04:31:50.132819912 +0100
@@ -65,7 +65,7 @@ module_param_named(modeset, vbox_modeset
 
 static struct drm_driver driver;
 
-static DEFINE_PCI_DEVICE_TABLE(pciidlist) =
+static const struct pci_device_id pciidlist[] =
 {
     {0x80ee, 0xbeef, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
     {0, 0, 0},
