--- ./Makefile.kmk.orig	2020-02-20 03:40:14.049126175 +0000
+++ ./Makefile.kmk	2020-02-20 03:40:41.843981024 +0000
@@ -188,15 +188,6 @@ ifeq ($(KBUILD_TARGET),darwin)
  					"$(if $(VBOX_WITH_HARDENING),/Applications/VirtualBox.app/Contents/Frameworks,@rpath)/$(qtmod).framework/Versions/5/$(qtmod)" ) ) \
  			"$(2)",) # HACK ALERT! Using $(source_type_prefix) to detect EXEC_SOURCES.
 else # win x11
- if1of ($(KBUILD_TARGET), linux solaris)
-  ifndef VBOX_ONLY_BUILD
-   ifneq ($(VBOX_GCC_VERSION_CXX),)
-    ifeq ($(int-ge $(VBOX_GCC_VERSION_CXX),40400),)
-     $(error gcc >= 4.4 required when compiling against Qt5!)
-    endif
-   endif
-  endif
- endif
  if defined(VBOX_WITH_ORACLE_QT) || defined(VBOX_WITH_QT_PAYLOAD)
   include $(KBUILD_PATH)/units/qt5.kmk
   ifeq ($(KBUILD_TARGET),win)
