Build drivers for XOrg 1.7 ABI

And only them.

diff -up VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxmouse/Makefile.kmk.xorg17 VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
--- VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxmouse/Makefile.kmk.xorg17	2009-11-05 17:54:27.000000000 +0100
+++ VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxmouse/Makefile.kmk	2009-11-12 14:27:16.000000000 +0100
@@ -61,7 +61,7 @@ endif
 #
 # vboxmouse_drv_70
 #
-DLLS += vboxmouse_drv_70
+#DLLS += vboxmouse_drv_70
 vboxmouse_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_70_DEFS = \
 	XFree86Server IN_MODULE XFree86Module XFree86LOADER XINPUT XORG_7X
@@ -82,7 +82,7 @@ vboxmouse_drv_70_SOURCES = \
 #
 # vboxmouse_drv_71
 #
-DLLS += vboxmouse_drv_71
+#DLLS += vboxmouse_drv_71
 vboxmouse_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_71_DEFS := $(vboxmouse_drv_70_DEFS)
 vboxmouse_drv_71_INCS := \
@@ -99,7 +99,7 @@ vboxmouse_drv_71_SOURCES = \
 #
 # vboxmouse_drv_14
 #
-DLLS += vboxmouse_drv_14
+#DLLS += vboxmouse_drv_14
 vboxmouse_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_14_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_14_INCS := \
@@ -116,7 +116,7 @@ vboxmouse_drv_14_SOURCES = \
 #
 # vboxmouse_drv_15
 #
-DLLS += vboxmouse_drv_15
+#DLLS += vboxmouse_drv_15
 vboxmouse_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_15_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_15_INCS := \
@@ -131,7 +131,7 @@ vboxmouse_drv_15_SOURCES = \
 #
 # vboxmouse_drv_16
 #
-DLLS += vboxmouse_drv_16
+#DLLS += vboxmouse_drv_16
 vboxmouse_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_16_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_16_INCS := \
@@ -150,15 +150,13 @@ vboxmouse_drv_16_SOURCES = \
 DLLS += vboxmouse_drv_17
 vboxmouse_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_17_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
-## @todo replace $(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local
 vboxmouse_drv_17_INCS := \
-	$(VBOX_PATH_X11_ROOT)/inputproto-1.9.99.902 \
-	$(VBOX_PATH_X11_ROOT)/libpciaccess-0.10.8 \
-	$(VBOX_PATH_X11_ROOT)/pixman-0.16.0 \
-	$(VBOX_PATH_X11_ROOT)/xextproto-7.1.1 \
-	$(VBOX_PATH_X11_ROOT)/xorg-server-1.6.99-20090831 \
-	$(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local \
-	$(VBOX_PATH_X11_ROOT)/xproto-7.0.15 \
+	/usr/include/pixman-1 \
+	/usr/share/xorg-x11-server-source/hw/xfree86/common \
+	/usr/share/xorg-x11-server-source/hw/xfree86/os-support/bus \
+	/usr/share/xorg-x11-server-source/include \
+	/usr/share/xorg-x11-server-source/mi \
+	/usr/share/xorg-x11-server-source/Xext \
 	$(PATH_SUB_CURRENT)
 vboxmouse_drv_17_SOURCES = \
 	vboxmouse_15.c
@@ -166,7 +164,7 @@ vboxmouse_drv_17_SOURCES = \
 
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
-ifdef VBOX_WITH_TESTCASES
+ifdef NOT_VBOX_WITH_TESTCASES
 # ifndef VBOX_ONLY_ADDITIONS
   if1of ($(KBUILD_TARGET),linux solaris)
    ifeq ($(KBUILD_HOST_ARCH),$(KBUILD_TARGET_ARCH))
diff -up VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk.xorg17 VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
--- VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk.xorg17	2009-11-10 16:47:42.000000000 +0100
+++ VirtualBox-3.1.0_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2009-11-12 14:29:08.000000000 +0100
@@ -113,7 +113,7 @@ endif   # target linux
 #         base keywords instead of using .solaris or .linux.
 #         Also it is *important* to use := and not = when deriving a properity.
 #
-DLLS += vboxvideo_drv_70
+#DLLS += vboxvideo_drv_70
 vboxvideo_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 if1of ($(KBUILD_TARGET), linux)
  vboxvideo_drv_70_CFLAGS += \
@@ -136,7 +136,7 @@ vboxvideo_drv_70_SOURCES  = \
 #
 # vboxvideo_drv_71
 #
-DLLS += vboxvideo_drv_71
+#DLLS += vboxvideo_drv_71
 vboxvideo_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_71_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_71_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -152,7 +152,7 @@ vboxvideo_drv_71_SOURCES  = \
 #
 # vboxvideo_drv_13
 #
-DLLS += vboxvideo_drv_13
+#DLLS += vboxvideo_drv_13
 vboxvideo_drv_13_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_13_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_13_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -174,7 +174,7 @@ vboxvideo_drv_13_SOURCES = \
 # This uses the same code (vboxvideo_13.c) as 1.3, but is built
 # for 1.4 as well in case there should be any relevant header changes.
 #
-DLLS += vboxvideo_drv_14
+#DLLS += vboxvideo_drv_14
 vboxvideo_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_14_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_14_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -193,7 +193,7 @@ vboxvideo_drv_14_SOURCES  = \
 #
 # vboxvideo_drv_15
 #
-DLLS += vboxvideo_drv_15
+#DLLS += vboxvideo_drv_15
 vboxvideo_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_15_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_15_DEFS := $(vboxvideo_drv_70_DEFS) NO_ANSIC PCIACCESS \
@@ -220,7 +220,7 @@ endif
 #
 # vboxvideo_drv_16
 #
-DLLS += vboxvideo_drv_16
+#DLLS += vboxvideo_drv_16
 vboxvideo_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_16_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_16_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -244,25 +244,38 @@ vboxvideo_drv_17_CFLAGS := $(vboxvideo_d
 vboxvideo_drv_17_DEFS := $(vboxvideo_drv_15_DEFS)
 ## @todo replace $(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local
 vboxvideo_drv_17_INCS = \
-	$(VBOX_PATH_X11_ROOT)/fontsproto-2.1.0 \
-	$(VBOX_PATH_X11_ROOT)/glproto-1.4.10 \
-	$(VBOX_PATH_X11_ROOT)/inputproto-1.9.99.902 \
-	$(VBOX_PATH_X11_ROOT)/libdrm-2.4.13 \
-	$(VBOX_PATH_X11_ROOT)/libpciaccess-0.10.8 \
-	$(VBOX_PATH_X11_ROOT)/pixman-0.16.0 \
-	$(VBOX_PATH_X11_ROOT)/randrproto-1.3.0 \
-	$(VBOX_PATH_X11_ROOT)/renderproto-0.11 \
-	$(VBOX_PATH_X11_ROOT)/xextproto-7.1.1 \
-	$(VBOX_PATH_X11_ROOT)/xf86driproto-2.1.0 \
-	$(VBOX_PATH_X11_ROOT)/xorg-server-1.6.99-20090831 \
-	$(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local \
-	$(VBOX_PATH_X11_ROOT)/xproto-7.0.15
+	/usr/include/drm \
+	/usr/include/X11/dri \
+	/usr/include/pixman-1 \
+	/usr/share/xorg-x11-server-source \
+	/usr/share/xorg-x11-server-source/fb \
+	/usr/share/xorg-x11-server-source/hw/xfree86/common \
+	/usr/share/xorg-x11-server-source/hw/xfree86/ddc \
+	/usr/share/xorg-x11-server-source/hw/xfree86/dixmods/extmod \
+	/usr/share/xorg-x11-server-source/hw/xfree86/dri \
+	/usr/share/xorg-x11-server-source/hw/xfree86/i2c \
+	/usr/share/xorg-x11-server-source/hw/xfree86/int10 \
+	/usr/share/xorg-x11-server-source/hw/xfree86/modes \
+	/usr/share/xorg-x11-server-source/hw/xfree86/os-support \
+	/usr/share/xorg-x11-server-source/hw/xfree86/os-support/bus \
+	/usr/share/xorg-x11-server-source/hw/xfree86/ramdac \
+	/usr/share/xorg-x11-server-source/hw/xfree86/shadowfb \
+	/usr/share/xorg-x11-server-source/hw/xfree86/vbe \
+	/usr/share/xorg-x11-server-source/hw/xfree86/vgahw \
+	/usr/share/xorg-x11-server-source/hw/xquartz/xpr \
+	/usr/share/xorg-x11-server-source/include \
+	/usr/share/xorg-x11-server-source/mi \
+	/usr/share/xorg-x11-server-source/miext/damage \
+	/usr/share/xorg-x11-server-source/randr \
+	/usr/share/xorg-x11-server-source/Xext \
+	/usr/share/xorg-x11-server-source/render \
+	/usr/share/xorg-x11-server-source/hw/xfree86/parser
 vboxvideo_drv_17_SOURCES := $(vboxvideo_drv_15_SOURCES)
 
 
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
-ifdef VBOX_WITH_TESTCASES
+ifdef NOT_VBOX_WITH_TESTCASES
 # ifndef VBOX_ONLY_ADDITIONS
   if1of ($(KBUILD_TARGET),linux solaris)
    ifeq ($(KBUILD_HOST_ARCH),$(KBUILD_TARGET_ARCH))
