Disable DRI for Rawhide kernel.

diff -up VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk.dri VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
--- VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk.dri	2009-08-04 18:24:00.033239741 -0400
+++ VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2009-08-04 18:59:37.438284750 -0400
@@ -203,7 +203,7 @@ vboxvideo_drv_14_SOURCES  = \
 vboxvideo_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_15_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_15_DEFS := $(vboxvideo_drv_70_DEFS) NO_ANSIC PCIACCESS \
-                         VBOX_NO_LOW_COLOUR VBOX_DRI
+                         VBOX_NO_LOW_COLOUR
 vboxvideo_drv_15_INCS = \
 	$(VBOX_PATH_X11_XORG_1_5) \
 	$(VBOX_PATH_X11_XORG_1_5)/X11 \
@@ -270,8 +270,8 @@ vboxvideo_drv_17_INCS = \
 	/usr/share/xorg-x11-server-source/hw/xfree86/parser
 vboxvideo_drv_17_SOURCES  = \
 	vboxvideo_15.c \
-	vboxutils.c \
-	vboxvideo_dri.c
+	vboxutils.c
+	#vboxvideo_dri.c
 
 ## Check the undefined symbols in the X.Org modules against lists of allowed
 ## symbols.  Not very elegant, but it will catch problems early.
