Defaults to Linux and to Fedora
Index: VirtualBox-6.0.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp
===================================================================
--- VirtualBox-6.0.0.orig/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp
+++ VirtualBox-6.0.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp
@@ -251,15 +251,15 @@ void UINameAndSystemEditor::sltFamilyCha
         if (iIndexWin7 != -1)
             m_pComboType->setCurrentIndex(iIndexWin7);
     }
-    /* Or select Oracle Linux item for Linux family as default: */
+    /* Or select Fedora item for Linux family as default: */
     else if (m_strFamilyId == "Linux")
     {
-        QString strDefaultID = "Oracle";
+        QString strDefaultID = "Fedora";
         if (ARCH_BITS == 64 && m_fSupportsHWVirtEx && m_fSupportsLongMode)
             strDefaultID += "_64";
-        const int iIndexUbuntu = m_pComboType->findData(strDefaultID, TypeID);
-        if (iIndexUbuntu != -1)
-            m_pComboType->setCurrentIndex(iIndexUbuntu);
+        const int iIndexFedora = m_pComboType->findData(strDefaultID, TypeID);
+        if (iIndexFedora != -1)
+            m_pComboType->setCurrentIndex(iIndexFedora);
     }
     /* Else simply select the first one present: */
     else
@@ -468,6 +468,10 @@ void UINameAndSystemEditor::prepareFamil
 
     /* Choose the 1st item to be the current: */
     m_pComboFamily->setCurrentIndex(0);
+    /* Default to Linux */
+    const int liIndex = m_pComboFamily->findData("Linux", TypeID);
+    if (liIndex != -1)
+        m_pComboFamily->setCurrentIndex(liIndex);
     /* And update the linked widgets accordingly: */
     sltFamilyChanged(m_pComboFamily->currentIndex());
 }
