--- VirtualBox-5.0.17-105882/src/VBox/Additions/linux/drm/vbox_mode.c.orig	2016-03-09 03:07:13.392279611 +0000
+++ VirtualBox-5.0.17-105882/src/VBox/Additions/linux/drm/vbox_mode.c	2016-03-09 03:08:22.875010971 +0000
@@ -401,7 +401,7 @@ static struct drm_encoder *vbox_encoder_
         return NULL;
 
     drm_encoder_init(dev, &vbox_encoder->base, &vbox_enc_funcs,
-             DRM_MODE_ENCODER_DAC);
+             DRM_MODE_ENCODER_DAC, NULL);
     drm_encoder_helper_add(&vbox_encoder->base, &vbox_enc_helper_funcs);
 
     vbox_encoder->base.possible_crtcs = 1 << i;
--- VirtualBox-5.0.17-105882/src/VBox/VMM/VMMR3/DBGFR3Trace.cpp.orig	2016-03-08 22:52:14.716735644 +0000
+++ VirtualBox-5.0.17-105882/src/VBox/VMM/VMMR3/DBGFR3Trace.cpp	2016-03-08 22:53:19.923458002 +0000
@@ -122,7 +122,7 @@ static int dbgfR3TraceEnable(PVM pVM, ui
 
     rc = RTTraceBufCarve(&hTraceBuf, cEntries, cbEntry, 0 /*fFlags*/, pvBlock, &cbBlock);
     AssertRCReturn(rc, rc);
-    AssertRelease(hTraceBuf == (RTTRACEBUF)pvBlock && (void *)hTraceBuf == pvBlock);
+    AssertRelease(hTraceBuf == (RTTRACEBUF)pvBlock);
 
     pVM->hTraceBufR3 = hTraceBuf;
     pVM->hTraceBufR0 = MMHyperCCToR0(pVM, hTraceBuf);
--- VirtualBox-5.0.17-105882/src/libs/kStuff/kStuff/kLdr/kLdrModMachO.c.orig	2016-03-08 20:25:40.831167519 +0000
+++ VirtualBox-5.0.17-105882/src/libs/kStuff/kStuff/kLdr/kLdrModMachO.c	2016-03-08 20:30:38.618274674 +0000
@@ -868,7 +868,7 @@ static int  kldrModMachOPreParseLoadComm
                         offSect += pSect->size; \
                     } while (0) /* VALIDATE_AND_ADD_SECTION */
 
-                    VALIDATE_AND_ADD_SECTION(32);
+                    //VALIDATE_AND_ADD_SECTION(32);
 
                     /* next */
                     pSect++;
@@ -920,7 +920,7 @@ static int  kldrModMachOPreParseLoadComm
                         pSect->reserved2 = K_E2E_U32(pSect->reserved2);
                     }
 
-                    VALIDATE_AND_ADD_SECTION(64);
+                    //VALIDATE_AND_ADD_SECTION(64);
 
                     /* next */
                     pSect++;
--- VirtualBox-5.0.16/src/VBox/Disassembler/DisasmCore.cpp	2016-03-04 19:28:26.000000000 +0000
+++ VirtualBox-5.0.17-105882/src/VBox/Disassembler/DisasmCore.cpp	2016-03-08 17:25:32.806645217 +0000
@@ -2532,7 +2532,8 @@ static size_t ParseVex3b(size_t offInstr
     uint8_t implOpcode = (byte1 & 0x1f);
 
     // REX.RXB
-    if (pDis->uCpuMode == DISCPUMODE_64BIT && ~(byte1 & 0xe0))
+    /** @todo Check this! was weird: ~(byte1 & 0xe0) */
+    if (pDis->uCpuMode == DISCPUMODE_64BIT && !(byte1 & 0xe0))
         pDis->fRexPrefix |= (byte1 >> 5) ^ 7;
 
     // VEX.W
