diff -urp VirtualBox-3.0.2_OSE.orig/src/VBox/Additions/x11/vboxvideo/vboxvideo_15.c VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/vboxvideo_15.c
--- VirtualBox-3.0.2_OSE.orig/src/VBox/Additions/x11/vboxvideo/vboxvideo_15.c	2009-07-10 11:40:23.000000000 -0400
+++ VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxvideo/vboxvideo_15.c	2009-08-04 15:54:37.137015652 -0400
@@ -475,6 +475,7 @@ static const xf86OutputFuncsRec VBOXOutp
     .destroy = vbox_output_stub
 };
 
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
 /*
  * List of symbols from other modules that this module references.  This
  * list is used to tell the loader that it is OK for symbols here to be
@@ -544,6 +545,7 @@ static const char *driSymbols[] = {
     NULL
 };
 #endif
+#endif
 
 #ifdef XFree86LOADER
 /* Module loader interface */
@@ -584,6 +586,7 @@ vboxSetup(pointer Module, pointer Option
 #else
         xf86AddDriver(&VBOXDRV, Module, 0);
 #endif
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
         LoaderRefSymLists(fbSymbols,
                           shadowfbSymbols,
                           vbeSymbols,
@@ -592,6 +595,7 @@ vboxSetup(pointer Module, pointer Option
                           drmSymbols, driSymbols,
 #endif
                           NULL);
+#endif
         return (pointer)TRUE;
     }
 
@@ -767,7 +771,9 @@ VBOXPreInit(ScrnInfoPtr pScrn, int flags
        text mode, in order to keep our code simple. */
     if (!xf86LoadSubModule(pScrn, "vbe"))
         return (FALSE);
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
     xf86LoaderReqSymLists(vbeSymbols, NULL);
+#endif
 
     if ((pVBox->pVbe = VBEExtendedInit(NULL, pVBox->pEnt->index,
                                        SET_BIOS_SCRATCH
@@ -787,16 +793,22 @@ VBOXPreInit(ScrnInfoPtr pScrn, int flags
     /* The ramdac module is needed for the hardware cursor. */
     if (!xf86LoadSubModule(pScrn, "ramdac"))
         return FALSE;
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
     xf86LoaderReqSymLists(ramdacSymbols, NULL);
+#endif
 
     /* The framebuffer module. */
     if (xf86LoadSubModule(pScrn, "fb") == NULL)
         return (FALSE);
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
     xf86LoaderReqSymLists(fbSymbols, NULL);
+#endif
 
     if (!xf86LoadSubModule(pScrn, "shadowfb"))
         return FALSE;
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
     xf86LoaderReqSymLists(shadowfbSymbols, NULL);
+#endif
 
     /* Set up our ScrnInfoRec structure to describe our virtual
        capabilities to X. */
@@ -906,7 +918,7 @@ VBOXPreInit(ScrnInfoPtr pScrn, int flags
     /* Framebuffer-related setup */
     pScrn->bitmapBitOrder = BITMAP_BIT_ORDER;
 
-#ifdef VBOX_DRI
+#if defined(VBOX_DRI) && GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 7
     /* Load the dri module. */
     if (xf86LoadSubModule(pScrn, "dri")) {
         xf86LoaderReqSymLists(driSymbols, drmSymbols, NULL);
