diff -Nurp VirtualBox-7.1.4.orig/configure VirtualBox-7.1.4/configure
--- VirtualBox-7.1.4.orig/configure	2023-07-12 19:25:04.000000000 +0300
+++ VirtualBox-7.1.4/configure	2023-07-19 22:10:00.941514360 +0300
@@ -2173,7 +2173,7 @@ extern "C" int main(void)
 }
 EOF
   found=
-  SUPPYTHONLIBS="python2.7 python2.6 python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.9 python3.10 python3.11 python3.12"
+  SUPPYTHONLIBS="python3.13 python3.13m python3.12 python3.12m python3.11 python3.11m python3.10 python3.10m python3.9 python3.9m python3.8 python3.8m python3.7 python3.7m"
   for p in $PYTHONDIR; do
     for d in $SUPPYTHONLIBS; do
       for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 lib/64 lib; do
