Index: /trunk/configure
===================================================================
--- trunk/configure	(revision 40412)
+++ trunk/configure	(revision 40476)
@@ -1961,5 +1961,5 @@
             cnf_append "VBOX_GSOAP_CXX_SOURCES" ""
           fi
-          cnf_append "VBOX_GSOAP_CXX_LIBS"    "libgsoap++"
+          cnf_append "VBOX_GSOAP_CXX_LIBS"    "libgsoapssl++ ssl crypto"
           log_success "found"
         else
Index: /trunk/src/VBox/Main/webservice/Makefile.kmk
===================================================================
--- trunk/src/VBox/Main/webservice/Makefile.kmk	(revision 40400)
+++ trunk/src/VBox/Main/webservice/Makefile.kmk	(revision 40476)
@@ -194,4 +194,9 @@
 
 
+# disable -fvisibility=hidden as the SOAP stuff does not properly set the visibility attributes
+TEMPLATE_VBOXWEBR3EXE = Webservices without -fvisibility
+TEMPLATE_VBOXWEBR3EXE_EXTENDS = VBOXR3EXE
+TEMPLATE_VBOXWEBR3EXE_CXXFLAGS = $(filter-out $(VBOX_GCC_fvisibility-hidden) $(VBOX_GCC_fvisibility-inlines-hidden),\
+				 $(TEMPLATE_VBOXR3EXE_CXXFLAGS))
 
 ifdef VBOX_GSOAP_INSTALLED
@@ -201,5 +206,5 @@
  #
  LIBRARIES += vboxsoap
- vboxsoap_TEMPLATE = VBOXR3EXE
+ vboxsoap_TEMPLATE = VBOXWEBR3EXE
  ifdef VBOX_USE_VCC80
   vboxsoap_CXXFLAGS.win += -bigobj
@@ -430,5 +435,5 @@
  #
  PROGRAMS += webtest
- webtest_TEMPLATE = VBOXR3EXE
+ webtest_TEMPLATE = VBOXWEBR3EXE
  ifdef VBOX_USE_VCC80
   webtest_CXXFLAGS.win += -bigobj
