--- ./src/libs/xpcom18a4/Config.kmk.c++17	2023-01-18 16:43:28.813456712 +0000
+++ ./src/libs/xpcom18a4/Config.kmk	2023-01-18 16:51:27.471594143 +0000
@@ -47,7 +47,7 @@ TEMPLATE_XPCOM_ASFLAGS             = $(N
 TEMPLATE_XPCOM_ASFLAGS.x86         = -m32
 TEMPLATE_XPCOM_ASFLAGS.amd64       = -m64
 TEMPLATE_XPCOM_ASDEFS              = $(NO_SUCH_VARIABLE)
-TEMPLATE_XPCOM_CXXFLAGS            = -g -pipe -ansi -Wall -Wno-unused -Wno-non-virtual-dtor \
+TEMPLATE_XPCOM_CXXFLAGS            = -g -pipe -ansi -Wall -Wno-unused -Wno-non-virtual-dtor -std=gnu++17 \
 	$(VBOX_GCC_Wno-invalid-offsetof) -Wno-sign-compare -Wno-unused -Wno-ctor-dtor-privacy \
 	$(VBOX_GCC_fvisibility-inlines-hidden) $(VBOX_GCC_fvisibility-hidden) \
 	$(VBOX_GCC_Wno-delete-non-virtual-dtor) $(VBOX_GCC_Wno-multistatement-macros) $(VBOX_GCC_fdiagnostics-show-option) \
