--- ./src/libs/xpcom18a4/Config.kmk.orig	2022-10-19 19:34:20.000000000 +0100
+++ ./src/libs/xpcom18a4/Config.kmk	2022-10-30 04:30:34.318046234 +0000
@@ -47,7 +47,7 @@ TEMPLATE_XPCOM_ASFLAGS             = $(N
 TEMPLATE_XPCOM_ASFLAGS.x86         = -m32
 TEMPLATE_XPCOM_ASFLAGS.amd64       = -m64
 TEMPLATE_XPCOM_ASDEFS              = $(NO_SUCH_VARIABLE)
-TEMPLATE_XPCOM_CXXFLAGS            = -g -pipe -ansi -Wall -Wno-unused -Wno-non-virtual-dtor \
+TEMPLATE_XPCOM_CXXFLAGS            = -g -pipe -ansi -Wall -Wno-unused -Wno-non-virtual-dtor -std=gnu++17 \
 	$(VBOX_GCC_Wno-invalid-offsetof) -Wno-sign-compare -Wno-unused -Wno-ctor-dtor-privacy \
 	$(VBOX_GCC_fvisibility-inlines-hidden) $(VBOX_GCC_fvisibility-hidden) \
 	$(VBOX_GCC_fdiagnostics-show-option) $(VBOX_GCC_Wno-delete-non-virtual-dtor) \
