Port to XInput 2.

Lubomir Rintel <lkundrak@v3.sk>

diff -urp VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxmouse/vboxmouse_15.c VirtualBox-3.0.2_OSE.fixed/src/VBox/Additions/x11/vboxmouse/vboxmouse_15.c
--- VirtualBox-3.0.2_OSE/src/VBox/Additions/x11/vboxmouse/vboxmouse_15.c	2009-07-10 11:40:23.000000000 -0400
+++ VirtualBox-3.0.2_OSE.fixed/src/VBox/Additions/x11/vboxmouse/vboxmouse_15.c	2009-08-04 14:58:36.353922261 -0400
@@ -75,10 +75,15 @@ VBoxInit(DeviceIntPtr device)
 {
     CARD8 map[2] = { 0, 1 };
     InputInfoPtr pInfo;
+    Atom axis_labels[2] = { 0, 0 };
+    Atom button_labels[2] = { 0, 0 };
 
     pInfo = device->public.devicePrivate;
 
     if (!InitValuatorClassDeviceStruct(device, 2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+                                       axis_labels,
+#endif
 #if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
                                        GetMotionHistory,
 #endif
@@ -86,15 +91,27 @@ VBoxInit(DeviceIntPtr device)
         return !Success;
 
     /* Pretend we have buttons so the server accepts us as a pointing device. */
-    if (!InitButtonClassDeviceStruct(device, 2 /* number of buttons */, map))
+    if (!InitButtonClassDeviceStruct(device, 2, /* number of buttons */
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+                                     button_labels,
+#endif
+                                     map))
         return !Success;
 
     /* Tell the server about the range of axis values we report */
-    xf86InitValuatorAxisStruct(device, 0, 0 /* min X */, 65536 /* max X */,
+    xf86InitValuatorAxisStruct(device, 0,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+                               axis_labels[0],
+#endif
+                               0 /* min X */, 65536 /* max X */,
                                10000, 0, 10000);
     xf86InitValuatorDefaults(device, 0);
 
-    xf86InitValuatorAxisStruct(device, 1, 0 /* min Y */, 65536 /* max Y */,
+    xf86InitValuatorAxisStruct(device, 1,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 7
+                               axis_labels[1],
+#endif
+                               0 /* min Y */, 65536 /* max Y */,
                                10000, 0, 10000);
     xf86InitValuatorDefaults(device, 1);
     xf86MotionHistoryAllocate(pInfo);
