--- VirtualBox-6.1.36/configure	2022-07-19 21:48:52.000000000 +0100
+++ VirtualBox-6.1.34.orig/configure	2022-07-25 14:00:03.679779468 +0100
@@ -2027,7 +2027,7 @@ extern "C" int main(void)
 }
 EOF
   found=
-  SUPPYTHONLIBS="python2.7 python2.6 python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.8m python3.9 python3.9m python3.10 python3.10m"
+  SUPPYTHONLIBS="python2.7 python2.6 python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.8m python3.9 python3.9m python3.10 python3.10m python3.11 python3.11m"
   for p in $PYTHONDIR; do
     for d in $SUPPYTHONLIBS; do
       for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 lib/64 lib; do
diff -rup VirtualBox-6.1.36/src/libs/xpcom18a4/python/gen_python_deps.py VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/gen_python_deps.py
--- VirtualBox-6.1.36/src/libs/xpcom18a4/python/gen_python_deps.py	2022-07-19 22:01:39.000000000 +0100
+++ VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/gen_python_deps.py	2022-07-25 14:00:04.027778063 +0100
@@ -16,7 +16,7 @@ from __future__ import print_function
 import os,sys
 from distutils.version import StrictVersion
 
-versions = ["2.6", "2.7", "3.1", "3.2", "3.2m", "3.3", "3.3m", "3.4", "3.4m", "3.5", "3.5m", "3.6", "3.6m", "3.7", "3.7m", "3.8", "3.8m", "3.9", "3.9m", "3.10", "3.10m" ]
+versions = ["2.6", "2.7", "3.1", "3.2", "3.2m", "3.3", "3.3m", "3.4", "3.4m", "3.5", "3.5m", "3.6", "3.6m", "3.7", "3.7m", "3.8", "3.8m", "3.9", "3.9m", "3.10", "3.10m", "3.11", "3.11m"]
 prefixes = ["/usr", "/usr/local", "/opt", "/opt/local"]
 known = {}
 
diff -rup VirtualBox-6.1.36/src/libs/xpcom18a4/python/Makefile.kmk VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/Makefile.kmk
--- VirtualBox-6.1.36/src/libs/xpcom18a4/python/Makefile.kmk	2022-07-19 22:01:39.000000000 +0100
+++ VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/Makefile.kmk	2022-07-25 14:00:03.971778289 +0100
@@ -26,7 +26,7 @@ endif
 
 #
 # List of supported Python versions, defining a number of
-# VBOX_PYTHON[26|27|31|32|32M|33|33M|34|34M|35|35M|36|36M|37|37M|38|38M|39|39M|310|310M|DEF]_[INC|LIB] variables
+# VBOX_PYTHON[26|27|31|32|32M|33|33M|34|34M|35|35M|36|36M|37|37M|38|38M|39|39M|310|310M|311|311M|DEF]_[INC|LIB] variables
 # which get picked up below.
 #
 ifeq ($(KBUILD_TARGET),darwin) # Relatively predictable, don't script.
@@ -624,6 +624,52 @@ VBoxPython3_10m_x86_LIBS       = $(VBOX_
   endif
  endif
 endif
+
+ifdef VBOX_PYTHON311_INC
+#
+# Python 3.11 version
+#
+DLLS += VBoxPython3_11
+VBoxPython3_11_EXTENDS    = VBoxPythonBase
+VBoxPython3_11_EXTENDS_BY = appending
+VBoxPython3_11_TEMPLATE   = XPCOM
+VBoxPython3_11_INCS       = $(VBOX_PYTHON311_INC)
+VBoxPython3_11_LIBS       = $(VBOX_PYTHON311_LIB)
+
+ ifdef VBOX_WITH_32_ON_64_MAIN_API
+  ifdef VBOX_PYTHON311_LIB_X86
+DLLS += VBoxPython3_11_x86
+VBoxPython3_11_x86_EXTENDS    = VBoxPythonBase_x86
+VBoxPython3_11_x86_EXTENDS_BY = appending
+VBoxPython3_11_x86_TEMPLATE   = XPCOM
+VBoxPython3_11_x86_INCS       = $(VBOX_PYTHON311_INC)
+VBoxPython3_11_x86_LIBS       = $(VBOX_PYTHON311_LIB_X86)
+  endif
+ endif
+endif
+
+ifdef VBOX_PYTHON311M_INC
+#
+# Python 3.11 version with pymalloc
+#
+DLLS += VBoxPython3_11m
+VBoxPython3_11m_EXTENDS    = VBoxPythonBase_m
+VBoxPython3_11m_EXTENDS_BY = appending
+VBoxPython3_11m_TEMPLATE   = XPCOM
+VBoxPython3_11m_INCS       = $(VBOX_PYTHON311M_INC)
+VBoxPython3_11m_LIBS       = $(VBOX_PYTHON311M_LIB)
+
+ ifdef VBOX_WITH_32_ON_64_MAIN_API
+  ifdef VBOX_PYTHON311M_LIB_X86
+DLLS += VBoxPython3_11m_x86
+VBoxPython3_11m_x86_EXTENDS    = VBoxPythonBase_x86_m
+VBoxPython3_11m_x86_EXTENDS_BY = appending
+VBoxPython3_11m_x86_TEMPLATE_  = XPCOM
+VBoxPython3_11m_x86_INCS       = $(VBOX_PYTHON311M_INC)
+VBoxPython3_11m_x86_LIBS       = $(VBOX_PYTHON311M_LIB_X86)
+  endif
+ endif
+endif
 
 ifdef VBOX_PYTHONDEF_INC
 #
diff -rup VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/module/_xpcom.cpp VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/module/_xpcom.cpp
--- VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/module/_xpcom.cpp	2022-07-19 22:01:40.000000000 +0100
+++ VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/module/_xpcom.cpp	2022-07-25 14:00:04.134777632 +0100
@@ -84,7 +84,11 @@ extern PYXPCOM_EXPORT void PyXPCOM_Inter
 #  define MANGLE_MODULE_INIT(a_Name)    RT_CONCAT(a_Name, MODULE_NAME_SUFFIX)
 # endif
 # ifdef VBOX_PYXPCOM_VERSIONED
-#  if   PY_VERSION_HEX >= 0x030a0000 && PY_VERSION_HEX < 0x030b0000
+#  if   PY_VERSION_HEX >= 0x030b0000 && PY_VERSION_HEX < 0x030c0000
+#   define MODULE_NAME    MANGLE_MODULE_NAME("VBoxPython3_11")
+#   define initVBoxPython MANGLE_MODULE_INIT(PyInit_VBoxPython3_11)
+
+#  elif   PY_VERSION_HEX >= 0x030a0000 && PY_VERSION_HEX < 0x030b0000
 #   define MODULE_NAME    MANGLE_MODULE_NAME("VBoxPython3_10")
 #   define initVBoxPython MANGLE_MODULE_INIT(PyInit_VBoxPython3_10)
 
diff -rup VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/PyXPCOM.h VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/PyXPCOM.h
--- VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/PyXPCOM.h	2022-07-19 22:01:40.000000000 +0100
+++ VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/PyXPCOM.h	2022-07-25 14:00:23.316663783 +0100
@@ -137,14 +137,6 @@ inline PyObject *PyBool_FromLong(long ok
 #  define PyInt_Check(o) PyLong_Check(o)
 #  define PyInt_AsLong(o) PyLong_AsLong(o)
 #  define PyNumber_Int(o) PyNumber_Long(o)
-#  if PY_VERSION_HEX <= 0x03030000 /* 3.3 added PyUnicode_AsUTF8AndSize */
-#   ifndef PyUnicode_AsUTF8
-#    define PyUnicode_AsUTF8(o) _PyUnicode_AsString(o)
-#   endif
-#   ifndef PyUnicode_AsUTF8AndSize
-#    define PyUnicode_AsUTF8AndSize(o,s) _PyUnicode_AsStringAndSize(o,s)
-#   endif
-#  endif
 typedef struct PyMethodChain
 {
     PyMethodDef *methods;
diff -rup VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/VariantUtils.cpp VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/VariantUtils.cpp
--- VirtualBox-6.1.36/src/libs/xpcom18a4/python/src/VariantUtils.cpp	2022-07-19 22:01:40.000000000 +0100
+++ VirtualBox-6.1.34.orig/src/libs/xpcom18a4/python/src/VariantUtils.cpp	2022-03-22 23:47:49.000000000 +0000
@@ -628,7 +628,7 @@ static PyObject *UnpackSingleArray(Py_ns
 #if PY_MAJOR_VERSION <= 2
 		return PyString_FromStringAndSize( (char *)array_ptr, sequence_size );
 #else
-		return PyBytes_FromStringAndSize( (char *)array_ptr, sequence_size );
+		return PyUnicode_FromStringAndSize( (char *)array_ptr, sequence_size );
 #endif
 
 	PRUint32 array_element_size = GetArrayElementSize(array_type);
