diff -up VirtualBox-2.1.4_OSE/configure.libcap VirtualBox-2.1.4_OSE/configure
--- VirtualBox-2.1.4_OSE/configure.libcap	2009-03-30 14:19:11.475691095 +0200
+++ VirtualBox-2.1.4_OSE/configure	2009-03-30 14:19:45.267690625 +0200
@@ -1548,6 +1548,7 @@ check_libcap()
   test_header "libcap library"
   cat > .tmp_src.cc << EOF
 #include <cstdio>
+#include <sys/types.h>
 #include <sys/capability.h>
 
 extern "C" int main(void)
