--- VirtualBox-5.0.17-105882/src/VBox/VMM/VMMR3/DBGFR3Trace.cpp.orig	2016-03-08 22:52:14.716735644 +0000
+++ VirtualBox-5.0.17-105882/src/VBox/VMM/VMMR3/DBGFR3Trace.cpp	2016-03-08 22:53:19.923458002 +0000
@@ -122,7 +122,7 @@ static int dbgfR3TraceEnable(PVM pVM, ui
 
     rc = RTTraceBufCarve(&hTraceBuf, cEntries, cbEntry, 0 /*fFlags*/, pvBlock, &cbBlock);
     AssertRCReturn(rc, rc);
-    AssertRelease(hTraceBuf == (RTTRACEBUF)pvBlock && (void *)hTraceBuf == pvBlock);
+    AssertRelease(hTraceBuf == (RTTRACEBUF)pvBlock);
 
     pVM->hTraceBufR3 = hTraceBuf;
     pVM->hTraceBufR0 = MMHyperCCToR0(pVM, hTraceBuf);
--- VirtualBox-5.0.17-105882/src/libs/kStuff/kStuff/kLdr/kLdrModMachO.c.orig	2016-03-08 20:25:40.831167519 +0000
+++ VirtualBox-5.0.17-105882/src/libs/kStuff/kStuff/kLdr/kLdrModMachO.c	2016-03-08 20:30:38.618274674 +0000
@@ -868,7 +868,7 @@ static int  kldrModMachOPreParseLoadComm
                         offSect += pSect->size; \
                     } while (0) /* VALIDATE_AND_ADD_SECTION */
 
-                    VALIDATE_AND_ADD_SECTION(32);
+                    //VALIDATE_AND_ADD_SECTION(32);
 
                     /* next */
                     pSect++;
@@ -920,7 +920,7 @@ static int  kldrModMachOPreParseLoadComm
                         pSect->reserved2 = K_E2E_U32(pSect->reserved2);
                     }
 
-                    VALIDATE_AND_ADD_SECTION(64);
+                    //VALIDATE_AND_ADD_SECTION(64);
 
                     /* next */
                     pSect++;
